classes/Telemetry.php

```diff
@@ -0,0 +1,353 @@
+		$response = wp_remote_post( 'https://api.wppopupmaker.com/wp-json/pmapi/v2/' . $action, array(
+			'method'      => 'POST',
+			'timeout'     => 20,
+			'redirection' => 5,
+			'httpversion' => '1.1',
+			'blocking'    => false,
+			'body'        => $data,
+			'user-agent'  => 'POPMAKE/' . Popup_Maker::$VER . '; ' . get_site_url(),
+		));
+
+		if ( is_wp_error( $response ) ) {
+			$error_message = $response->get_error_message();
+			PUM_Utils_Logging::instance()->log( sprintf( 'Cannot send telemetry data. Error received was: %s', esc_html( $error_message ) ) );
+			return false;
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Adds admin notice if we haven't asked before.
+	 *
+	 * @param array $alerts The alerts currently in the alert system.
+	 * @return array Alerts for the alert system.
+	 * @since 1.11.0
+	 */
+	public static function optin_alert( $alerts ) {
+		if ( ! self::should_show_alert() ) {
+			return $alerts;
+		}
+
+		$optin_url = add_query_arg( 'pum_optin_check', 'optin' );
+
+		ob_start();
+		?>
+		<ul>
+			<li><a href="<?php echo esc_attr( $optin_url ); ?>"><strong><?php esc_html_e( 'Allow', 'popup-maker' ); ?></strong></a></li>
+			<li><a href="#" class="pum-dismiss"><?php esc_html_e( 'Do not allow', 'popup-maker' ); ?></a></li>
+			<li><a href="https://docs.wppopupmaker.com/article/528-the-data-the-popup-maker-plugin-collects" target="_blank" rel="noreferrer noopener"><?php esc_html_e( 'Learn more', 'popup-maker' ); ?></a></li>
+		</ul>
+		<?php
+		$html = ob_get_clean();
+		$alerts[] = array(
+			'code'        => 'pum_telemetry_notice',
+			'type'        => 'info',
+			'message'     => esc_html__( "Allow Popup Maker to track this plugin's usage and help us make this plugin better? No user data is sent to our servers. No sensitive data is tracked.", 'popup-maker' ),
+			'html'        => $html,
+			'priority'    => 10,
+			'dismissible' => true,
+			'global'      => false,
+		);
+		return $alerts;
+	}
+
+	/**
+	 * Checks if any options have been clicked from admin notices.
+	 *
+	 * @since 1.11.0
+	 */
+	public static function optin_alert_check() {
+		if ( isset( $_GET['pum_optin_check'] ) ) {
+			if ( 'optin' === $_GET['pum_optin_check'] ) {
+				pum_update_option( 'telemetry', true );
+			}
+		}
+	}
+
+	/**
+	 * Whether or not we should show optin alert
+	 *
+	 * @since 1.11.0
+	 * @return bool True if alert should be shown
+	 */
+	public static function should_show_alert() {
+		return false === self::has_opted_in() && current_user_can( 'manage_options' ) && false == get_option( '_pum_telemetry_notice_dismissed', false );
+	}
+
+	/**
+	 * Determines if it is time to send telemetry data.
+	 * @return bool True if it is time.
+	 * @since 1.11.0
+	 */
+	public static function is_time_to_send() {
+
+		// Only send if admin has opted in.
+		if ( ! self::has_opted_in() ) {
+			return false;
+		}
+
+		// Send a maximum of once per week.
+		if ( get_transient( 'pum_tracking_last_send' ) ) {
```

Comment:
Should this use a transient which can be cleared, or an option that stores timestamp of last send and we compare if that is > 7 days from now.